File: R/linear-regression.r
library(ggplot2)
#load data
data <- read.csv("Downloads/MachineLearning-master/Example Data/OLS_Regression_Example_3.csv", stringsAsFactors=F))
#replace gender with 0 and 1, male and female respectively
data$Gender[which(data$Gender=="Male")] = 0
data$Gender[which(data$Gender=="Female")] = 1
#change us metrics to eu matrics
data$Height = data$Height * 2.54
data$Weight = data$Weight * 0.45359237
#plot data (human)
human.plot <- ggplot(data)
human.plot + geom_point(aes(x = Height, y = Weight)) + labs(title = "Weight and heights for male and females")
#plot data (female and male separate)
maleFemale.plot <- ggplot(data)
maleFemale.plot + geom_point(aes(x = Height, y = Weight, color = Gender)) + labs(title = "Weight and heights for male and females")
#train a linear model
lm_model <- lm(Weight~Gender+Height, data=data)
summary(lm_model)
#predict a test data
test.data <- data.frame(0, 170)
colnames(test.data) <- c("Gender", "Height")
test.data$Gender <- as.factor(test.data$Gender)
predict(lm_model, test.data)

File: Graphlab/linear_regression.py
# coding: utf-8
# ###Linear Regression with Graphlab 

import graphlab

#load data
data = graphlab.SFrame.read_csv("Downloads/MachineLearning-master/Example Data/OLS_Regression_Example_3.csv")

data.head()
data.tail()

#Set visualize target to ipython notebook
graphlab.canvas.set_target("ipynb")

#Replace Us metrics with cm and kilo's
height_transform = lambda x:x * 2.54
weight_transform = lambda x:x * 0.45359237


data['Height'] = height_transform(data['Height'])
data['Weight'] = weight_transform(data['Weight'])

#view data
data.show(view="Scatter Plot", x = "Height", y = "Weight")

#train a linear model
lm_model = graphlab.linear_regression.create(data, target="Weight", features=['Gender', 'Height'])

#we can get that rmse = 4.54
#predict weight based on male, height 170, female 170
#firstly construct a SFrame
predict_data = graphlab.SFrame({'Gender':['Male','Female'], "Height":[170,170]})


lm_model.predict(predict_data)



